core/src/env_platform/db_handler.bal

```diff
@@ -10,76 +10,98 @@ mongodb:Collection userCollection = config_handler:getUserCollection();
-
-    boolean result = check saveApplicationMetadata(form.title);
-    log:printDebug("Saved information in application metadata: " + result.toString());
-    // Construct the application.
-    map<json> application = {
-        "applicationId": check generateApplicationId(form.applicationCreatedDate, form.title),
-        "status": form.status,
-        "numberOfVersions": 1,
-        "title": form.title,
-        "versions": [
-                {
-                    "applicationCreatedDate": {
-                        "year": form.applicationCreatedDate.year,
-                        "month": form.applicationCreatedDate.month,
-                        "day": form.applicationCreatedDate.day,
-                        "hour": form.applicationCreatedDate.hour,
-                        "minute": form.applicationCreatedDate.minute
-                    },
-                    "removalDate": {
-                        "year": form.removalDate.year,
-                        "month": form.removalDate.month,
-                        "day": form.removalDate.day,
-                        "hour": form.removalDate.hour,
-                        "minute": form.removalDate.minute
-                    },
-                    "reason": form.reason,
-                    "applicationType": form.applicationType,
-                    "requestedBy": form.requestedBy,
-                    "permitRequired": form.permitRequired,
-                    "landOwner": form.landOwner,
-                    "treeRemovalAuthority": form.treeRemovalAuthority,
-                    "city": form.city,
-                    "district": form.district,
-                    "nameOfTheLand": form.nameOfTheLand,
-                    "planNumber": form.planNumber,
-                    "area": extractAreaAsJSONArray(form.area),
-                    "treeInformation": extractTreeInformationAsJSONArray(form.treeInformation)
-                }
-            ]
-    };
-    log:printDebug("Constructed application: " + application.toString());
-
-    mongodb:DatabaseError? inserted = applicationCollection->insert(application);
-
-    if (inserted is mongodb:DatabaseError) {
-        log:printDebug("An error occurred while saving the application with ID: " + application.applicationId.toString() + ". " + inserted.reason().toString() + ".");
+# + userId - Id of the User. 
+# + return - Returns [true, applicationId] if the application is saved, error if there is a mongodb:DatabaseError or  
+# if there's an error while generating the applicationId.
+function saveApplication(TreeRemovalForm form, string userId) returns [boolean, string]|error {
+
+    if (check isValidUser(userId)) {
+        string applicationId = check generateApplicationId(form.applicationCreatedDate, form.title);
+
+        // Added form information to metadata.
+        boolean result = check saveApplicationMetadata(form.title);
+        log:printDebug("Saved information in application metadata: " + result.toString());
```

Comment:
Any explanation for this change @sanjulamadurapperuma?

Reply:
Yes waiting for the explanation @sanjulamadurapperuma

Reply:
This change was to make it more legible, otherwise reading the message being printed in the logs give the impression that the information has been saved successfully. Any concerns with that? Or is the switch from **:** to **=>** bothering you? 

If that's the case, then we can leave the colon, but we will have to rephrase the log message. It is not clear to those who are not familiar with the flow.

Reply:
I am okay with the explanation. What about you @deshankoswatte

Reply:
I'm ok with that too.